<launch>
    <arg
        name="run_node"
        default="true"
        doc="Run the panoramagrid_node node. Can be set to &quot;false&quot;, mainly for development purposes."
    />
    <arg
        name="gazebo"
        default="false"
    />
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find panoramagrid)/config/rosconsole.conf"/>

    <node
        pkg="rviz"
        type="rviz"
        name="rviz"
        args="-d $(find panoramagrid)/config/default.rviz -f world"
        required="true"
        output="screen"
    />
    <node
        pkg="panoramagrid"
        type="node"
        name="panoramagrid_node"
        required="true"
        output="screen"
        if="$(arg run_node)"
    >
        <param name="path" value="/usr/local/share/panoramagrid/grid2_lighter.zip" />
        <param name="global_frame" value="world" />
        <param name="gazebo" value="$(arg gazebo)" />
        <param name="image_topic" value="image" />
    </node>
    <node
        pkg="panoramagrid"
        type="node"
        name="panoramagrid_node2"
        required="true"
        output="screen"
        if="$(arg run_node)"
    >
        <param name="path" value="/usr/local/share/panoramagrid/grid2_20cm_lighter.zip" />
        <param name="global_frame" value="world" />
        <param name="gazebo" value="$(arg gazebo)" />
        <param name="image_topic" value="image2" />
    </node>
    <include if="$(arg gazebo)" file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find panoramagrid)/worlds/demo.world"/>
    </include>
    <include if="$(arg gazebo)" file="$(find rcprg_gazebo_utils)/launch/gazebo_publish_ros_tf_object.launch">
        <arg name="link_name" value="camera::link"/>
	    <arg name="frame_id" value="camera"/>
    </include>
</launch>
